src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/ConnectionHandlerContext.cs

```diff
@@ -31,5 +31,40 @@ internal class ConnectionHandlerContext : HandlerRequest
         /// Class used by orchestrator while chaining handlers.
         /// </summary>
         public Exception Error { get; set; }
+
+        /// <summary>
+        /// Method needed to clone ConnectionHandlerContext as part of history along the chain.
+        /// </summary>
+        public object Clone() 
+        {
+            ConnectionHandlerContext clonedContext = new ConnectionHandlerContext();
+
+            if (this.ConnectionStream != null)
+            {
+                MemoryStream clonedStream = new MemoryStream();
+                this.ConnectionStream.CopyTo(clonedStream);
+                clonedStream.Position = 0;
+                clonedContext.ConnectionStream = clonedStream;
+            }
```

Comment:
Is this correct? I expect `ConnectionStream` would ultimately be based on a `NetworkStream`. Particularly for read-only routing, I would have thought we'd only want to clone the connection details and let the handlers re-establish the connection.

Reply:
Yes, this isn't required anymore.